src/Translator/TranslationScanner.php

```diff
@@ -54,14 +54,12 @@ private function getFileContent(string $filePath): string
      */
     private function getKeysFromFunction(string $functionName, string $content): array
     {
-        preg_match_all("#{$functionName}\((.*?)\)#", $content, $matches);
+        preg_match_all("#{$functionName} *\( *((['\"])((?:\\\\\\2|.)*?)\\2)#", $content, $matches);
 
         $matches = $matches[1] ?? [];
 
         return array_reduce($matches, function (array $keys, string $match) {
-            preg_match("#(['\"])((?:\\\\\\1|.)*?)\\1#", $match, $matchKey);
-
-            $key = $matchKey[2] ?? '';
+            eval("\$key = $match;");
```

Comment:
`eval` is not safe and could potentially make this an unsafe library, please check if `$key = stripcslashes(trim($match, '\'\"'));` solves this issue

Reply:
The problem I'm having currently is that I have a string like this (slightly modified from Laravel's default email template): `"If you’re having trouble clicking the \":actionText\" button, copy and paste the URL below\ninto your web browser: [:actionURL](:actionURL)"`. Now, if we scan it and write out as JSON, the backslashes and double quotes will be escaped, so we end up with `"If you’re having trouble clicking the \\\":actionText\\\" button, copy and paste the URL below\\ninto your web browser: [:actionURL](:actionURL)"`, which is not what we want. The string should stay the same, since it's a double quoted string originally. But if we had something similar in single quotes, the behavior would be different - then we would actually need the backslashes escaped.

`stripcslashes` certainly doesn't help in this situation. I think we need to have separate logic for single and double quoted strings. The logic needs to match PHP's evaluation of the strings exactly, as only then will the keys match the actual strings passed to the translation function. That's why I went with `eval` as this is the simplest to use, but I am aware using it is considered bad practice for obvious reasons.

Reply:
Avoided using `eval` by using `stripcslashes` for double quoted strings, and simply replacing escaped single quotes and backslashes otherwise, which should mimic the behavior of PHP exactly.